### PR TITLE
Feat/state batch max txs

### PR DESCRIPTION
**Pull request by karlfloersch**
**Description**
Make it possible to config both tx batch size & state batch size.

**Additional context**
State root batches should be much larger (eg. 2k) than tx batches

### CODE REVIEW

**Comment by tynes** [comment]
This is a backwards incompatible change, what do you think about making the env vars be:

- `MAX_TX_BATCH_COUNT`
- `MAX_STATE_BATCH_COUNT`

Since these correspond to the number of elements and there is a `MAX_TX_SIZE` which corresponds to the size in bytes

**Comment by karlfloersch** [comment]
What do you think about the changes to the env vars that I just made? Included your suggestion but also made the tx size env var more explicit as well.

**Comment by tynes** [comment]
I think the `IN` is a bit unnecessary, `MIN_L1_TX_SIZE_BYTES` is more standard in my experience

**Comment by karlfloersch** [comment]
@tynes do you think I should change this to `_BYTES` or is `_IN_BYTES` fine or just remove the `BYTES` suffix altogether? Is there a particular place where you see this `_BYTES` convention used?

**Comment by tynes** [comment]
> do you think I should change this to _BYTES or is _IN_BYTES fine or just remove the BYTES suffix altogether? Is there a particular place where you see this _BYTES convention used?

I think it was fine without the `BYTES` suffix, what is the most important is that its consistent across all of the services. Examples in the wild where words like in and bytes are not used:
- https://github.com/handshake-org/hsd/blob/a69780e4060b154ab45527c40973bde75129295c/lib/wallet/node.js#L51
- https://github.com/ethereum/go-ethereum/blob/c5df05b9a9529cb17b4416e23cd24583335c6a9d/cmd/utils/flags.go#L374

Both refer to mb and are not explicit. Something that I think would be really nice to standardize around would be similar to this: https://pkg.go.dev/github.com/inhies/go-bytesize - where you would say `MIN_L1_TX_SIZE: 100B` and the library would convert the string `100B` to 100 (bytes). In the short term I think we should change the minimum possible and come up with a plan around how all config should be standardized

**Comment by karlfloersch** [comment]
Nice agreed

File: src/exec/run-batch-submitter.ts [hunk 1]
```diff
@@ -164,9 +167,9 @@ export const run = async () => {
   const stateBatchSubmitter = new StateBatchSubmitter(
     sequencerSigner,
     l2Provider,
-    parseInt(requiredEnvVars.MIN_TX_SIZE, 10),
-    parseInt(requiredEnvVars.MAX_TX_SIZE, 10),
-    parseInt(requiredEnvVars.MAX_BATCH_SIZE, 10),
+    parseInt(requiredEnvVars.MIN_L1_TX_SIZE, 10),
+    parseInt(requiredEnvVars.MAX_L1_TX_SIZE, 10),
+    parseInt(requiredEnvVars.MAX_TX_BATCH_COUNT, 10),
```
Comment (tynes) [inline src/exec/run-batch-submitter.ts:172]: `MAX_STATE_BATCH_COUNT`

**Review by tynes — APPROVED** [review]

**Pull request merged**